Backport a build fix on x86_64 (8bfd04b974689f700bbd053ad6e66b0a95fb80c9)

From: Jeff Dike <jdike@linux.intel.com>

x86_64 defines either memcpy or __memcpy depending on the gcc version, and
it looks like UML needs to follow that in its exporting.

Cc: Gabriel C <nix.or.die@googlemail.com>
Signed-off-by: Jeff Dike <jdike@linux.intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Index: linux-source-2.6.25/arch/um/sys-x86_64/ksyms.c
===================================================================
--- linux-source-2.6.25.orig/arch/um/sys-x86_64/ksyms.c	2008-07-24 07:02:07.607642178 +0900
+++ linux-source-2.6.25/arch/um/sys-x86_64/ksyms.c	2008-07-24 07:10:42.959643438 +0900
@@ -13,4 +13,8 @@ EXPORT_SYMBOL(__down_failed_trylock);
 EXPORT_SYMBOL(__up_wakeup);
 
 /*XXX: we need them because they would be exported by x86_64 */
+#if (__GNUC__ == 4 && __GNUC_MINOR__ >= 3) || __GNUC__ > 4
+EXPORT_SYMBOL(memcpy);
+#else
 EXPORT_SYMBOL(__memcpy);
+#endif
